src/classes/HiFiCommunicator.ts

```diff
@@ -392,11 +556,7 @@ export class HiFiCommunicator {
      * is still in the process of initializing its underlying HiFiMixerSession).
      */
     getConnectionState(): HiFiConnectionStates {
-        if (this._mixerSession) {
-            return this._mixerSession.getCurrentHiFiConnectionState();
-        } else {
-            return null;
-        }
+        return this._currentHiFiConnectionState;
```

Comment:
@howard-stearns This is where getConnectionState() lives nowadays. I just moved the tracking of the current connection state up a level to the communicator, so that the communicator could more easily make decisions based on it.